testonly/hammer/reader.go

```diff
@@ -0,0 +1,145 @@
+package hammer
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math/rand"
+	"reflect"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/google/trillian"
+	"github.com/google/trillian/client"
+	"github.com/google/trillian/testonly"
+	"github.com/google/trillian/types"
+)
+
+type intSelector func(prng *rand.Rand) int
+
+// validReadOps performs valid read operations against the map.
```

Comment:
Are there invalid ones? Maybe rename to `readOps`?

Reply:
There will be invalid ones! See performInvalidOp on hammer.go. The plan is to have one interface for performing readOps, and two implementations (one that is valid, and one that is invalid). We could even have a third implementation which randomly chooses between valid vs invalid, which then sucks even more of the logic of how to do things out of the main hammer.

This invalidReadOps will be the next PR.